akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala

```diff
@@ -1435,6 +1435,7 @@ private[stream] object Collect {
     private var groupEmitted = true
     private var finished = false
     private var totalWeight = 0L
+    private var elementCount = 0L
```

Comment:
Can't come up with something more clever. I agree a boolean would be more clear as it is either empty or nonEmpty